File: output/cs/ifc2x3_cartesianpoint.cs

/*
{
    "$id": [
        "ifc2x3",
        "cartesianpoint"
    ],
    "$schemaversion": "1",
    "$comment": "TODO: fix",
    "description": "Cartesianpoint of ifc version 2x3",
    "shape": {
        "points": {
            "type": "array",
            "items": {
                "type": "number"
            }
        },
        "cardinality": {
            "type": "number"
        }
    }
}
*/

using bimrepo;

namespace ifc2x3 {
    // TODO: fix
    public class cartesianpoint : ECSComponent {
        
        public cartesianpoint() : base("ifc2x3_cartesianpoint") {
            
        }
        
        // properties
        public List<int> points = new();
            public int cardinality = 0;
            // end properties
            
            // methods
            bool validate() {
                return false;
            }
            
            
            public override ComponentT exportToDataArray() {
                ComponentT componentObj = new();
                
                componentObj.Type = new List<string>() {"ifc2x3","cartesianpoint"};
                    componentObj.Data = new List<ComponentDataT>();
                        
                        
                        // property points
                        componentObj.Data.Add(Helper.MakeArrayStart(this.points.Count));
                        this.points.ForEach((item) => componentObj.Data.Add(Helper.MakeNumber(item)));
                        componentObj.Data.Add(Helper.MakeArrayEnd());
                        
                        
                        // property cardinality
                        componentObj.Data.Add(Helper.MakeNumber(this.cardinality));
                        
                        
                        return componentObj;
                        }
                        
                        
                        public override SchemaT exportDefinitionToArray() {
                            var schemaObj = new SchemaT();
                            schemaObj.Id = new List<string>() { "ifc2x3","cartesianpoint" };
                                schemaObj.Schemaversion = "1";
                                schemaObj.Comment = "TODO: fix";
                                schemaObj.Description = "Cartesianpoint of ifc version 2x3";
                                
                                
                                schemaObj.SchemaShape = new shapeT();
                                
                                schemaObj.SchemaShape.Properties = new List<propertyT>();
                                    
                                    
                                    // property points
                                    {
                                        var prop_points = new propertyT();
                                        
                                        prop_points.Name = "points";
                                        prop_points.Type = PropertyType.Array;
                                        
                                        prop_points.Items = new ItemsT();
                                        prop_points.Items.Type = PropertyType.Number;
                                        
                                        
                                        schemaObj.SchemaShape.Properties.Add(prop_points);
                                    }
                                    
                                    
                                    // property cardinality
                                    {
                                        var prop_cardinality = new propertyT();
                                        
                                        prop_cardinality.Name = "cardinality";
                                        prop_cardinality.Type = PropertyType.Number;
                                        
                                        
                                        schemaObj.SchemaShape.Properties.Add(prop_cardinality);
                                    }
                                    
                                    
                                    
                                    return schemaObj;
                                    }
                                    
                                    // end methods
                                    
                                    // statics
                                    
                                    
                                    public static cartesianpoint importFromDataArray(ComponentT componentObj) {
                                        // TODO: check if component type matches the class
                                        
                                        var obj = new cartesianpoint();
                                        
                                        
                                        // property points
                                        {
                                            var count = Helper.GetArrayStart(componentObj);
                                            for (var i = 0; i < count; i++)
                                            {
                                                obj.points.Add(Helper.GetNumber(componentObj));
                                            }
                                            Helper.Expect(componentObj, ComponentDataType.ArrayEnd);
                                        }
                                        
                                        
                                        // property cardinality
                                        {
                                            obj.cardinality = Helper.GetNumber(componentObj);
                                        }
                                        
                                        
                                        return obj;
                                    }
                                    
                                    // end statics
                                    }
                                    }

File: output/ts/ifc2x3_cartesianpoint.ts

/*
{
    "$id": [
        "ifc2x3",
        "cartesianpoint"
    ],
    "$schemaversion": "1",
    "$comment": "TODO: fix",
    "description": "Cartesianpoint of ifc version 2x3",
    "shape": {
        "points": {
            "type": "array",
            "items": {
                "type": "number"
            }
        },
        "cardinality": {
            "type": "number"
        }
    }
}
*/

import * as flatbuffers from 'flatbuffers';
import { ECSComponent } from '../../ecs';
import { Component, ComponentDataType, ComponentT, ItemsT, PropertyType, SchemaT, propertyT, shapeT } from '../../bimrepo';
import { Expect, GetArrayStart, GetNumber, MakeArrayEnd, MakeArrayStart, MakeNumber } from '../../helper';

export namespace ifc2x3 {
    // TODO: fix
    export class cartesianpoint extends ECSComponent {
        
        constructor() {
            super("ifc2x3_cartesianpoint");
        }
        
        // properties
        points: number[] = [];
        cardinality: number = 0
        // end properties
        
        // methods
        validate(): boolean {
            return false;
        }
        
        
        exportToDataArray(): ComponentT {
            let componentObj = new ComponentT();
            
            componentObj.type = ["ifc2x3","cartesianpoint"];
            
            
            // property points
            componentObj.data.push(MakeArrayStart(this.points.length));
            this.points.forEach((item) => componentObj.data.push(MakeNumber(item)));
            componentObj.data.push(MakeArrayEnd());
            
            
            // property cardinality
            componentObj.data.push(MakeNumber(this.cardinality));
            
            
            return componentObj;
        }
        
        
        exportDefinitionToArray(): SchemaT {
            let schemaObj = new SchemaT();
            schemaObj.id = ["ifc2x3","cartesianpoint"];
            schemaObj.schemaversion = "1";
            schemaObj.comment = "TODO: fix";
            schemaObj.description = "Cartesianpoint of ifc version 2x3";
            
            
            schemaObj.schemaShape = new shapeT();
            
            
            // property points
            {
                let prop_points = new propertyT();
                
                prop_points.name = "points";
                prop_points.type = PropertyType.Array;
                
                prop_points.items = new ItemsT();
                prop_points.items.type = PropertyType.Number;
                
                
                schemaObj.schemaShape.properties.push(prop_points);
            }
            
            
            // property cardinality
            {
                let prop_cardinality = new propertyT();
                
                prop_cardinality.name = "cardinality";
                prop_cardinality.type = PropertyType.Number;
                
                
                schemaObj.schemaShape.properties.push(prop_cardinality);
            }
            
            
            
            return schemaObj;
        }
        
        // end methods
        
        // statics
        
        
        static importFromDataArray(componentObj: ComponentT): cartesianpoint {
            // TODO: check if component type matches the class
            
            let obj = new cartesianpoint();
            
            
            // property points
            {
                let count = GetArrayStart(componentObj);
                for (let i = 0; i < count; i++)
                {
                    obj.points.push(GetNumber(componentObj));
                }
                Expect(componentObj, ComponentDataType.ArrayEnd);
            }
            
            
            // property cardinality
            {
                obj.cardinality = GetNumber(componentObj);
            }
            
            
            return obj;
        }
        
        // end statics
    }
}